### ISSUE TITLE
DB migration error

### ISSUE THREAD

**Issue by atcoco** [catarse/catarse#2890]
Hi, everyone! Nice to meet you. I'm a rails developer interested in this project.

To setup local dev environment, I have some issues.
When I run rake db:migrate, I get this error:

`PG::UndefinedTable: ERROR:  relation "common_schema.subscriptions" does not exist`

I noticed this error is caused by 20171205174414_add_more_fdw_views.rb.
In the migration file, the env is set as test. So I tried rake db:migrate RAILS_ENV=test.
And it worked.

Any idea about my issue?

Regards

**Comment by adrianob** [comment]
Hello!
You need to setup foreign data wrapper before running this migration by running ```rake common:generate_fdw```.
More details available here https://comum.gitbooks.io/docs/content/local-development-setup/catarse-integration-setup.html
We will update the docs shortly.

**Comment by atcoco** [comment]
Thanks for your explanation.
I'll try it. 👍

**Comment by bestsuperweb** [comment]
Hi Adrianob, Nice to meet you. I encountered this error with migrate.
So i have run `rake common:generate_fdw`, btw there is an error:
`PG::SyntaxError: ERROR:  syntax error at or near "common_db"
LINE 8:         SERVER common_db`
Could you let me know what is the solution.

**Comment by adrianob** [comment]
@bestsuperweb you probably haven't set up the necessary settings. 
```CatarseSettings[:common_db_host] = 'localhost'
CatarseSettings[:common_db_name] = 'service_core'
CatarseSettings[:common_db_port] = '5444'

CatarseSettings[:fdw_user] = 'read_user_fdw'
CatarseSettings[:common_db_user] = 'read_user_fdw'
CatarseSettings[:common_db_password] = 'example_fdw'```
Of course, adapt them to your environment. If you don't intend to have subscription projects I would recommend to use an older version for now, as the setup is quite complex at the moment. We are moving the whole system to docker to easy deployments off but it's not quite ready yet.

**Comment by bestsuperweb** [comment]
Thanks for adrianob,
Could you let me know old version url?

**Comment by adrianob** [comment]
@bestsuperweb I've create a new tag with the latest stable version here https://github.com/catarse/catarse/releases/tag/3.0

**Comment by bestsuperweb** [comment]
Thanks, @adrianob

**Comment by kashiftufail** [comment]
Hi @adrianob,

I have set these values on Rails console

      CatarseSettings[:common_db_name] = 'service_core'
      CatarseSettings[:common_db_port] = '5444'

       CatarseSettings[:fdw_user] = 'read_user_fdw'
       CatarseSettings[:common_db_user] = 'read_user_fdw'
       CatarseSettings[:common_db_password] = 'example_fdw'

when i  run 
           rake common:generate_fdw

I have this error

         PG::UndefinedObject: ERROR:  role "read_user_fdw" does not exist

on this command 

              $ rake db:create db:migrate db:seed
This error

               PG::UndefinedTable: ERROR: relation "common_schema.subscriptions" does not exist

Any help to run project successfully 

Thanks

**Comment by adrianob** [comment]
@kashiftufail those are just example values, the role needs to exist in the first place.
If you are using the default configs in https://github.com/common-group/services-core-db then these settings should work out of the box:
```
CatarseSettings[:fdw_user] = 'postgres'
CatarseSettings[:common_db_user] = 'postgres'
CatarseSettings[:common_db_port] = '5432'
CatarseSettings[:common_db_password] = 'example'
CatarseSettings[:common_db_name] = 'service_core'
```

**Comment by kashiftufail** [comment]
@licolsan it is just dummy values. I am trying to run catarse on my local  from 2 days but not successfull. Instruction are not well defined in Readme file.

**Comment by licolsan** [comment]
Yeah, me too. And I don't know where the "catarse" role come from?

`PG::UndefinedObject: ERROR:  role "catarse" does not exist`

**Comment by licolsan** [comment]
I fix my problem by adding role "catarse" to postgresql db

**Comment by istehbabunqa** [comment]
Hello, @adrianob:

(1) Documentation for install is not updated with this one yet. When is that expected ? This is urgent/critical for first installs

(2) You want everyone to "manually" enter these settings in rake console everytime they run this app ? Wouldn't it have been better to have and read these settings in/from a config file ?

**Comment by adrianob** [comment]
@istehbabunqa documentation is indeed severely outdated right now. When we finish migrating everything to the new services-core-db all these FDW steps won't be necessary anymore, but for now A PR would be welcomed to integrate @licolsan instructions into the README. 

For production environment you will need at least 2 db instances. One for the rails app and another one for services-core-db. FDW is used to connect the rails db to the services-core-db while we haven't finished migrating. If you don't plan on having subscription projects you'd probably be better off using an older version for now.

**Comment by teddyswu** [comment]
Same problem, But i add 
CatarseSettings[:fdw_user] = 'postgres'
CatarseSettings[:common_db_user] = 'postgres'
CatarseSettings[:common_db_port] = '5432'
CatarseSettings[:common_db_password] = 'example'
CatarseSettings[:common_db_name] = 'service_core'

not work.........

Is there any other solution?

Thanks~

**Comment by rraallvv** [comment]
Has anyone been able to install Catarse from this repository with the latest changes?
